applications/ecallmgr/src/ecallmgr_fs_bridge.erl

```diff
@@ -68,42 +70,41 @@ unbridge(UUID, JObj) ->
 %% Bridge command helpers
 %% @end
 %%--------------------------------------------------------------------
+-spec handle_ringback(atom(), ne_binary(), wh_json:object()) -> 'ok'.
 handle_ringback(Node, UUID, JObj) ->
-    case wh_json:get_value(<<"Ringback">>, JObj) of
-        'undefined' ->
-            case wh_json:get_value([<<"Custom-Channel-Vars">>, <<"Ringback">>], JObj) of
-                'undefined' -> 'ok';
-                Media ->
-                    Stream = ecallmgr_util:media_path(Media, 'extant', UUID, JObj),
-                    lager:debug("bridge has custom ringback in channel vars: ~s", [Stream]),
-                    ecallmgr_util:set(Node, UUID, [{<<"ringback">>, Stream}])
-            end;
+    case wh_json:get_first_defined([<<"Ringback">>
+                                    ,[<<"Custom-Channel-Vars">>, <<"Ringback">>]
+                                   ]
+                                   ,JObj
+                                  )
+    of
+        'undefined' -> 'ok';
         Media ->
-            Stream = ecallmgr_util:media_path(Media, extant, UUID, JObj),
+            Stream = ecallmgr_util:media_path(Media, 'extant', UUID, JObj),
             lager:debug("bridge has custom ringback: ~s", [Stream]),
             ecallmgr_util:set(Node, UUID, [{<<"ringback">>, Stream}])
     end.
 
+-spec maybe_early_media(atom(), ne_binary(), wh_json:object()) -> 'ok'.
 maybe_early_media(Node, UUID, JObj) ->
-    case wh_json:get_value(<<"Dial-Endpoint-Method">>, JObj, <<"single">>) of
-        <<"simultaneous">> ->
-            case length(wh_json:get_ne_value(<<"Endpoints">>, JObj, [])) > 1 of
-                'false' -> 'ok';
-                'true' ->
-                    lager:debug("bridge is simultaneous to multiple endpoints, starting local ringing"),
-                    %% we don't really care if this succeeds, the call will fail later on
-                    ecallmgr_util:send_cmd(Node, UUID, <<"ring_ready">>, "")
-            end;
-        _Else -> 'ok'
+    Endpoints = wh_json:get_ne_value(<<"Endpoints">>, JObj, []),
+    case ecallmgr_util:get_dial_separator(JObj, Endpoints) of
```

Comment:
Previously it sent ring_ready when trying more than 1 simultaneous endpoint.  The refactor now appears to send it if its single?